src/upgrades/mod.rs

```diff
@@ -769,6 +777,29 @@ pub fn prepare_upgrade_data_migrations(
                 pre_0_14_0::UpgradeAggregateStoreTrustAnchorProxy::upgrade(TA_PROXY_SERVER_NS, mode, config)?;
 
                 Ok(Some(UpgradeReport::new(aspa_configs, true, versions)))
+            } else if versions.from < KrillVersion::release(0, 14, 3) {
+                // Check all possibly affected stores for corrupted files resulting
+                // from issue #1160 and fix them if needed.
```

Comment:
This comment, to me at least, is misleading. Nothing will be "fixed", if problematic store items are found they are deleted which to me is quite different than fixing them.

Reply:
Correction: Deletion only happens in the `upgrade_tasks()` and `upgrade_status()` calls, I see that the `upgrade_agg()` call for example does attempt a sort of fix via replay (though replay as far as it can which presumably is also "deleting" the bits that it can't).